### ISSUE TITLE
Invalid @inheritDocs handle

### ISSUE THREAD

**Issue by pawelkg** [vimeo/psalm#2199]
For some reason code similar to this one:
https://psalm.dev/r/ee9aac2363
Throws error on my code:
```
RROR: MissingThrowsDocblock - src\Drivers\Output\OutputXMLWriter.php:39:9 - InvalidArgumentException is thrown but not caught - please either catch or add a @throws annotation
        parent::__construct($config);
```
I need to add additional @throws. I use dev-master (commit 9b31483).
File I test with this error: https://github.com/wszetko/sitemap/blob/master/src/Drivers/Output/OutputXMLWriter.php

**Comment by muglug** [comment]
Reproduction: https://psalm.dev/r/aefb7ec0bb

**Comment by muglug** [comment]
This actually isn't a bug – `@inheritDocs` is automatic, but constructors don't follow normal rules of inheritance (as a constructor can have different parameters from its parent).

`@throws` inheritance works fine here:

```php
<?php
  
abstract class A
{
  /**
   * Class constructor. 
   *
   * @param array $test
   * @throws InvalidArgumentException
   */
  public function foo($test) : void
  {
    if (!isset($test['domain'])) {
      throw new InvalidArgumentException('Domain is not set.');
    }
  }
}

class C extends A
{  
  /**
   * @inheritDoc
   */
  public function foo($test) : void
  {
    parent::foo($test);
  }
}
```

https://psalm.dev/r/15c1bec947
